Cargo.toml

```diff
@@ -30,6 +30,7 @@ siphasher = "1"
 thiserror = "1.0.63"
 tokio = { version = "1.40.0", features = ["fs", "macros", "sync", "rt", "rt-multi-thread"] }
 ulid = { version = "1.1.3", features = ["serde"] }
+uuid = { version = "1.11.0", features = ["v4", "serde"] }
```

Comment:
I don't see any discussion about id formats in the RFC, so apologies if this was already discussed. What's the reason to use UUIDs instead of ULIDs for checkpoints? Wouldn't be better to keep a consistent ID format across all features?

Reply:
I suppose. The only property that I really cared about was for the checkpoint IDs to be unique so it felt most intuitive to just use a UUID. The main benefit I can see is that it's a more "standard" type that's baked into most languages, so it's easy to use (e.g. no additional deps required) if someone wanted to for example write a python or Java program to inspect checkpoints. That said, we're using ULIDs for SST names anyway so I don't think this is a big deal. So I'm on board with using a ULID for consistency's sake. Any strong opinions @vigneshc @criccomini ?

Reply:
UUID is more readily available in some languages (C# for example). 
Would we support scenarios that lets the caller pass in the checkpoint id? That could be interesting for some use cases. For example, a lookup identifier is created externally and is used to store multiple entities, including slatedb. 
If we support such cases, UUID has a slight benefit.

Otherwise, we can keep it as ulid.

Reply:
IIRC, there was some discussion about exposing the checkpoint in the range scan iterator. I kinda feel like UUID is the way to go, but I don't feel strongly about it. Mostly just s compatibility thing when exposing public APIs. I suppose we could expose ULIDs as strings, though.

Reply:
let’s go with UUIDs, it does sound like the way to go.